### PR TITLE
feat: Implement `TablePrimaryCell` and `TableCellActions`

### LINKED ISSUE #24225: Table: Create primary column primitive

### PR DESCRIPTION
**Pull request by ling1726**
Fixes #24225
Addresses #23983
msedge_EXoz5LFCjr
msedge_iQ7IllMRS5

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `192.699 kB`<br />`52.786 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `32.688 kB`<br />`10.736 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/66762e1ca870a21db46b72c9b0848c5b107b8e09'>66762e1ca870a21db46b72c9b0848c5b107b8e09</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/Avatar-mount_base.html">1275</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/Avatar-mount.html">1293</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/Button-mount_base.html">939</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/Button-mount.html">933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProvider-mount_base.html">1483</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProvider-mount.html">1475</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-mount_base.html">566</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-mount.html">573</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">537</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">535</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">570</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/MakeStyles-mount_base.html">1989</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/MakeStyles-mount.html">1978</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/SpinButton-mount_base.html">2431</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24232/perf-test-react-components/SpinButton-mount.html">2414</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-table/src/components/TablePrimaryCell/TablePrimaryCell.types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { TableCellSlots } from '../TableCell/TableCell.types';
+
+export type TablePrimaryCellSlots = {
```
Comment (bsunderhus) [inline packages/react-components/react-table/src/components/TablePrimaryCell/TablePrimaryCell.types.ts:4]: root?

**Review by bsunderhus — APPROVED** [review]
LGTM!

File: packages/react-components/react-table/src/components/TablePrimaryCell/TablePrimaryCell.types.ts (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-table/src/components/TablePrimaryCell/TablePrimaryCell.types.ts:4]: I've extended from `TableCell`

**Pull request merged**